### PR TITLE
feat: add config properties descriptions [sc-0]

### PR DESCRIPTION
**Pull request by nahuelon**
I hereby confirm that I followed the code guidelines found at [engineering guidelines](https://www.notion.so/checkly/Engineering-Guidelines-a3a165a203a04dc1a112f0e26b2f2d3f)

### Notes for the reviewer
- Add configuration properties descriptions
- Remove duplicated properties and now `CheckConfigDefaults` picks properties from the base `interface CheckProps`

### CODE REVIEW

File: package/src/constructs/api-check.ts [hunk 1]
```diff
@@ -17,16 +17,30 @@ enum BodyType {
...
+
 interface BasicAuth {
   username: string
   password: string
 }
 
 export interface Request {
   url: string,
-  method: string,
+  method: HttpRequestMethod,
```
Comment (kiroushi) [inline package/src/constructs/api-check.ts:37]: I would use a union literal type here, is much more ergonomic from a user perspective. Even Axios does it that way: https://github.com/axios/axios/blob/18c1710f244f54d5891a017714bb58f21f5039bd/index.d.ts#L191

File: package/src/services/checkly-config-loader.ts [hunk 2]
```diff
@@ -1,36 +1,40 @@
...
+   * Checks default configuration properties.
+   */
   checks?: CheckConfigDefaults & {
-    locations?: string[],
-    runtimeId?: string,
     checkMatch?: string,
     ignoreDirectoriesMatch?: string[],
     browserChecks?: CheckConfigDefaults & {
       testMatch?: string,
     },
```
Comment (kiroushi) [inline package/src/services/checkly-config-loader.ts:33]: Nit: it would be good to have a description on these too.

**Review by kiroushi — APPROVED** [review]
Looks neat 👍

File: package/src/constructs/alert-channel.ts [hunk 3]
```diff
@@ -1,13 +1,36 @@
...
+  sslExpiryThreshold: number
+}
 
+/**
+ * Creates an Alert Channels
+ *
+ * @remarks
+ *
+ * This class make use of the Alert Channels endpoints
+ * listed {@link https://developers.checklyhq.com/reference/postv1alertchannels here}
```
Comment (umutuzgur) [inline package/src/constructs/alert-channel.ts:32]: This is not true. We have different schemas for these constructs. I would not put the public api docs here. We can add the link when we put project/deploy behind v1 as well

Comment (umutuzgur) [reply package/src/constructs/alert-channel.ts]: Sounds good 👍

**Review by umutuzgur — APPROVED** [review]

**Pull request merged**